### PR TITLE
Add documentation on data flows for analytics.

### PR DESCRIPTION
**Pull request by benkags**

### CODE REVIEW

**Comment by benkags** [comment]
@garethbowen added this here but not sure if this is the best place to have it. Please advise accordingly if not.  cc @michaelkohn

File: development/data-flows-for-analytics.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+![overview](img/data-flows-overview-1.png)
+
+![overview](img/data-flows-overview-2.png)
+
+At a high level:
+
+- data is collected from the device of a health worker;
+- this data is pushed to an online instance from where this data is available to other health workers, supervisors and decision makers;
+- the data is transferred to a relational database and presented in forms suitable for analytics and visualizations;
```
Comment (garethbowen) [inline development/data-flows-for-analytics.md]:
```suggestion
- the data is transferred to a relational database and made available analytics and visualizations;
```

The word "forms" is overloaded as it's used to mean other things.

File: development/data-flows-for-analytics.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+- more access to the data is given to relevant parties at this level, for example analysts;
+- managers and decision makers access the same from visualizations of the data.
```
Comment (garethbowen) [inline development/data-flows-for-analytics.md]:
```suggestion
- managers and decision makers access the same visualizations of the data.
```

File: development/data-flows-for-analytics.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+- access control management is left to the partner technical teams where possible;
+- dashboard data source refresh intervals set to align with project needs;
+- update of the data sources monitored to ensure updating works as expected;
+- review of the dashboards as part of the design process;
+- qualitative design activities to interrogate trends observed in the dashboards and iterate on them if need be;
+
+### Beyond current pipeline
+
+[cht-core](https://github.com/medic/cht-core) is mostly data collection tools and is the first in the data management pipeline. It is the core part of a deployment but the rest of the tools can be easily replaced with other preferred options. It also helps that couch2pg is an open source tool which provides the opportunity for collaboration to extend its functionality to support other implementations. Klipfolio, the tool that we currently use for visualizations, is a proprietary tool but there exist open source options such as [Apach Superset](https://superset.incubator.apache.org/) that are worth exploring.
```
Comment (garethbowen) [inline development/data-flows-for-analytics.md]:
```suggestion
[cht-core](https://github.com/medic/cht-core) is mostly data collection tools and is the first in the data management pipeline. It is the core part of a deployment but the rest of the tools can be easily replaced with other preferred options. It also helps that couch2pg is an open source tool which provides the opportunity for collaboration to extend its functionality to support other implementations. Klipfolio, the tool that we currently use for visualizations, is a proprietary tool but there exists open source options such as [Apache Superset](https://superset.incubator.apache.org/) that are worth exploring.
```

File: development/data-flows-for-analytics.md (hunk 1 above)
Comment (michaelkohn) [reply development/data-flows-for-analytics.md]: I'd probably give couch2pg a plug here, to make it clear that it requires a separate component.

File: development/data-flows-for-analytics.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+![overview](img/data-flows-overview-1.png)
+
+![overview](img/data-flows-overview-2.png)
+
+At a high level:
+
+- data is collected from the device of a health worker;
+- this data is pushed to an online instance from where this data is available to other health workers, supervisors and decision makers;
+- the data is transferred to a relational database and presented in forms suitable for analytics and visualizations;
+- more access to the data is given to relevant parties at this level, for example analysts;
```
Comment (michaelkohn) [inline development/data-flows-for-analytics.md]: I think you are talking about giving access to PostgreSQL here, for clarity I think we should say so.

File: development/data-flows-for-analytics.md (hunk 2 above)
Comment (michaelkohn) [inline development/data-flows-for-analytics.md]: Maybe include a line stating that visualization platforms can then be connected to PostgreSQL

File: development/data-flows-for-analytics.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+Ultimately all the data ends up in a CouchDB instance deployed in the cloud whether through data synchronization with PouchDB local to the health workers devices, use of SMS aggregators or gateway. It should be mentioned that you can have a deployment supported by all of webapp, mobile app and SMS and have all the data end up in the same CouchDB instance.
+
+#### 2. Data transformation
+
+We use a service called couch2pg to move data from CouchDB to a relational database, PostgreSQL in this case. The choice of PostgreSQL for analytics dashboard data sources is to allow use of the more familiar SQL querying. It is an open source tool that can be [easily deployed](https://github.com/medic/medic-couch2pg#installation-steps-if-applicable). When deployed the service uses [CouchDB's changes feed](https://docs.couchdb.org/en/2.2.0/api/database/changes.html) which allows capturing of everything happening in CouchDB in incremental updates. It is run and monitored by the operating system where it is configured to fetch data at a configurable interval.
```
Comment (michaelkohn) [inline development/data-flows-for-analytics.md]: Should link out to https://github.com/medic/medic-couch2pg

**Comment by benkags** [comment]
I've addressed the comments & suggestions. Please have a second look. @michaelkohn @garethbowen

**Comment by garethbowen** [comment]
There's one typo still - "Apach Superset" should be "Apache Superset".

File: development/data-flows-for-analytics.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+Data in the views and functions mentioned in this section is as accurate as the accuracy of the SQL queries. Best practice is to begin the process of defining these objects at design in order to align analytics and dashboards requirements with workflows being deployed.
+
+#### 3. Data use
+
+The data in PostgreSQL is mostly either used by direct querying or via dashboard visualizations. Database visualizations are built scoped to the requirements of supporting a successful deployment. In-depth data analysis is supported at the PostgreSQL level through updated contactviews, formviews, useviews and functions with access to these provided to relevant parties as and when needed.
+
+As mentioned previously, formviews are built to present data in a structure similar to the data collection tool (form) used. Useviews are tailored to align with a use case, mostly using the formviews are the data sources. These are fundamentally guided by design of the workflows and should be interpreted in the context of the design materials including a document explaining the definitions of variables used.
```
Comment (michaelkohn) [inline development/data-flows-for-analytics.md]:
```suggestion
As mentioned previously, formviews are built to present data in a structure similar to the data collection tool (form) used. Useviews are tailored to align with a use case, mostly using the formviews as the data sources. These are fundamentally guided by design of the workflows and should be interpreted in the context of the design materials including a document explaining the definitions of variables used.
```

File: development/data-flows-for-analytics.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+[cht-core](https://github.com/medic/cht-core) is mostly data collection tools and is the first in the data management pipeline. It is the core part of a deployment but the rest of the tools can be easily replaced with other preferred options. It also helps that couch2pg is an open source tool which provides the opportunity for collaboration to extend its functionality to support other implementations. Klipfolio, the tool that we currently use for visualizations, is a proprietary tool but there exists open source options such as [Apach Superset](https://superset.incubator.apache.org/) that are worth exploring.
+
+## Backup
+
+The machines running each of CouchDB and PostgreSQL instances are backed up daily.
+
+## Summary
+
+The diagram below summarizes the details in the document.
```
Comment (michaelkohn) [inline development/data-flows-for-analytics.md]: I would actually probably leave this diagram out (ie remove it). It's pretty specific to our usage, includes very custom components that were manually created (dwh_impact, for example), and represents a structure that we are actively looking to replace.

File: development/data-flows-for-analytics.md (hunk 6 above)
Comment (benkags) [reply development/data-flows-for-analytics.md]: updated

File: development/data-flows-for-analytics.md (hunk 7 above)
Comment (benkags) [reply development/data-flows-for-analytics.md]: Done.

**Pull request merged**